src/utilities/tests/mounted-ref.test.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import {mount} from '@shopify/react-testing';
+
+import {useMountedRef} from '../mounted-ref';
+
+describe('useMountedRef()', () => {
+  it('returns a ref with current value as true when the component is mounted', () => {
+    const spy = jest.fn((_) => {});
+
+    function MockComponent() {
+      const mounted = useMountedRef();
+      spy(mounted.current);
+      return null;
+    }
+
+    mount(<MockComponent />);
+    expect(spy).toHaveBeenCalledWith(true);
+  });
+
+  it('returns a ref with current value as false when the component is un-mounted', async () => {
```

Comment:
This is the same test used in @shopify/react-hooks except we are getting a warning because of the promise.

`Warning: The callback passed to ReactTestUtils.act(...) function must not return anything.`

Any suggestions on how we can test that unmounting returns false without it?

Reply:
Got this to work, thoughts?

`
it('returns a ref with current value as false when the component is un-mounted', () => {
    const spy = jest.fn((_) => {});

    function MockComponent() {
      const mounted = useMountedRef();

      useEffect(() => {
        return () => {
          // eslint-disable-next-line react-hooks/exhaustive-deps
          spy(mounted.current);
        };
      }, [mounted]);

      return <div />;
    }

    const mockComponent = mount(<MockComponent />);

    mockComponent.unmount();

    expect(spy).toHaveBeenCalledWith(false);
  });
`

Reply:
Are our type files up to date?